connect/runtime/src/main/java/org/apache/kafka/connect/runtime/errors/LogReporter.java

```diff
@@ -69,4 +69,7 @@ String message(ProcessingContext context) {
                 context.toString(connConfig.includeRecordDetailsInErrorLog()));
     }
 
+    @Override
+    public void close() {
```

Comment:
want to add this as a default implementation in the ErrorReporter interface?

Reply:
It might save us a few LOC if we added a default implementation but I kind of like the guarantee that any new implementation of the `ErrorReporter` interface would have to explicitly add a `close` method since it might help prevent resource leaks similar to the ones that warranted these changes in the future. And it's not public API, so we don't have to worry about breaking compatibility by adding a new method with no default implementation.

Reply:
I'll revert this to reduce the number of files changed and preserve backwards compatibility, but I would be fine with either strategy.

When implementing this method for the LogReporter, I did for a moment have to scan the file and consider what resources might need to be cleaned up. I appreciate that the non-default method forced me to do that, and when/if another ErrorReporter subclass is written, I think it will be helpful too.